### DISCUSSION TITLE
Greener days

### DISCUSSION THREAD

**Discussion by xlibbyx** [BottlecapDave/HomeAssistant-OctopusEnergy#661] [Ideas]
I try to use the greener days website to choose which evenings to plug my EV in. I don't think this is available on the API, but would it be possible to get the information from the web page and display it in HA?

Website: https://octopus.energy/smart/greener-days/

**Comment by BottlecapDave** [comment]
Hello. If the data isn't available via the API then it won't be possible to expose the information in the integration. You could use the HA scrape integration to try and scrape the information. Alternatively, this information is probably based on data provided by the national grid. In which case you could use the carbon intensity integration in HACS to get this information (I also developed a variation myself which is pinned to my profile)

**Reply by BottlecapDave** [reply]
There's rumblings in the Octopus Energy forums that this could come in the future via the API...

**Comment by kevinmckeown** [comment]
Time passed and it's been added (yet another "thanks Dave!"), though disabled.  To enable, try HA / Settings / Integrations / Octopus Energy / 87 entities, search for 'green', click on entity(s) you want, cog, enable.  Then get distracted by other disabled entities that you didn't know you needed ;)

See https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/greenness_forecast/
and https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq/#there-are-entities-that-are-disabled-why-are-they-disabled-and-how-do-i-enable-them

Or if you just want to see the data without logging into your account: https://energy-stats.uk/octopus-greener-days/
